src/devices/bus/isa/myb3k_fdc.cpp

```diff
@@ -0,0 +1,376 @@
+}
+
+//-------------------------------------------------
+//  drq_w - signal dma request to ISA bus
+//-------------------------------------------------
+WRITE_LINE_MEMBER( isa8_myb3k_fdc4710_device::drq_w )
+{
+	LOG("%s: %d\n", FUNCNAME, state);
+	m_isa->drq2_w(state ? ASSERT_LINE : CLEAR_LINE);
+}
+
+WRITE_LINE_MEMBER( isa8_myb3k_fdc4711_device::drq_w )
+{
+	LOG("%s: %d\n", FUNCNAME, state);
+	m_isa->drq2_w(state ? ASSERT_LINE : CLEAR_LINE);
+}
+
+//-------------------------------------------------
+//  dack_r - return FDC data trough DMA
+//-------------------------------------------------
+uint8_t isa8_myb3k_fdc4710_device::dack_r(int line)
+{
+	return m_fdc->data_r();
+}
+
+uint8_t isa8_myb3k_fdc4711_device::dack_r(int line)
+{
+	return m_fdc->data_r();
+}
+
+//-------------------------------------------------
+//  dack_w - write DMA data to FDC
+//-------------------------------------------------
+void isa8_myb3k_fdc4710_device::dack_w(int line, uint8_t data)
+{
+	return m_fdc->data_w(data);
+}
+
+void isa8_myb3k_fdc4711_device::dack_w(int line, uint8_t data)
+{
+	return m_fdc->data_w(data);
+}
+
+#if 0 // eop/tc is used to for logic around multi sector transfers
+void isa8_myb3k_fdc4710_device::eop_w(int state)
+{
+	m_fdc->tc_w(state == ASSERT_LINE);
+}
+
+void isa8_myb3k_fdc4711_device::eop_w(int state)
+{
+	m_fdc->tc_w(state == ASSERT_LINE);
+}
+#endif
+
+//--------------------------------------------------------
+//  myb3k_inv_fdc_data_r - a LS240 inverts databus for FDC
+//--------------------------------------------------------
+READ8_MEMBER( isa8_myb3k_fdc4710_device::myb3k_inv_fdc_data_r )
+{
+	uint8_t tmp = m_fdc->read(space, offset);
+	LOGR("%s: %02x -> %02x\n", FUNCNAME, tmp, (~tmp) & 0xff);
+	return ~tmp;
+}
+
+READ8_MEMBER( isa8_myb3k_fdc4711_device::myb3k_inv_fdc_data_r )
+{
+	uint8_t tmp = m_fdc->read(space, offset);
+	LOGR("%s: %02x -> %02x\n", FUNCNAME, tmp, (~tmp) & 0xff);
+	return ~tmp;
+}
+
+//--------------------------------------------------------
+//  myb3k_inv_fdc_data_w - a LS240 inverts databus for FDC
+//--------------------------------------------------------
+WRITE8_MEMBER( isa8_myb3k_fdc4710_device::myb3k_inv_fdc_data_w )
+{
+	LOG("%s: %02x -> %02x\n", FUNCNAME, data, (~data) & 0xff);
+	m_fdc->write(space, offset, (~data) & 0xff);
+}
+
+WRITE8_MEMBER( isa8_myb3k_fdc4711_device::myb3k_inv_fdc_data_w )
+{
+	LOG("%s: %02x -> %02x\n", FUNCNAME, data, (~data) & 0xff);
+	m_fdc->write(space, offset, (~data) & 0xff);
+}
+
+//-------------------------------------------------
+//  myb3k_fdc_command - descrete fdc card features
+//-------------------------------------------------
+#define FDC_MSM_MODE   0x40
+#define FDC_DDEN       0x20
+//#define FDC_MOTOR_ON   0x10 // According to service manual but not schematics and BIOS
+#define FDC_SIDE_SEL   0x08
+#define FDC_MOTOR_ON   0x04 // According to schematics but "Motor Cont" according to service manual
+#define FDC_DRIVE_SEL  0x03
```

Comment:
If these are only going to be used in one function, please use local `constexpr` or enum or something.  Preprocessor macros can't be scoped and produce unhelpful error messages when things go wrong, and hence should be avoided if something else will do the same job.

Reply:
ok, I'll add it to the upcoming update of Mybrain 3000

Reply:
By the way @cuavas , as you can see the classes for the two ISA card in this file are 99% the same code, only the fdc device type and the number of floppies differ. I could use some C++ hints how to create a base class with all the methods and only keep the last 1% in the derived classes, I tried but the exercise is giving me grave headaches. I will add the 8" fdc as well when I get hands on one and I would hate to tripple the code...

I figure the memory map, the fdc device and the floppy configuration will differ